langchain/agents/mrkl/prompt.py

```diff
@@ -4,7 +4,7 @@
 
 Question: the input question you must answer
 Thought: you should always think about what to do
-Action: the action to take, should be one of [{tool_names}]
+Action: the action to take, should be one of [{tool_names}, None]
```

Comment:
hmmm why is this neccesary?

Reply:
if I use `gpt-3.5` without that addition the notebook fails complaining that `Python REPL` tool does not exist. I think there might be a problem with parsing a `single` tool. I could try to debug further why that happens.

Reply:
yeah that would be interesting to look into. seems a bit janky to add None

Reply:
Fair enough, I'll spend more time debugging that behavior and report

Reply:
I retested it and if I remove that ugly ` None` section, the agent can't find the `Python REPL` tool and goes in a loop with:

```Observation: [Python REPL] is not a valid tool, try another one.```

Now let's see if I can debug why things go wrong.